SQL.txt

```diff
@@ -0,0 +1,1451 @@
+Empty set (0.00 sec)
+
+16) Select all orders that had amounts that were greater than at least one of the orders from October 6. 
+
+
+mysql> SELECT * FROM ORDERS WHERE AMT> ANY(SELECT AMT FROM ORDERS WHERE ODATE='1990-10-06');
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3002 |  1900.1 | 1990-10-03 | 2007 | 1004 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3008 |    4723 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
++------+---------+------------+------+------+
+
+17) Write a query that uses the EXISTS operator to extract all salespeople who have customers with a rating of 300.
+ 
+
+mysql> SELECT * FROM SALESPEOPLE WHERE EXISTS (SELECT * FROM CUSTOMERS WHERE SALESPEOPLE.SNUM=CUSTOMERS.SNUM AND RATING =300);
++------+--------+-----------+------+
+| SNUM | SNAME  | CITY      | COMM |
++------+--------+-----------+------+
+| 1002 | Serres | SanJose   |   13 |
+| 1007 | Rofkin | Barcelona |   15 |
++------+--------+-----------+------+
+
+ 18) Find all customers whose cnum is 1000 above the snum of Serres. 
+
+mysql> SELECT * FROM CUSTOMERS WHERE CNUM>=(SELECT 1000 + SNUM FROM SALESPEO
+PLE WHERE SNAME ='SERRES');
++------+----------+---------+--------+------+
+| CNUM | CNAME    | CITY    | RATING | SNUM |
++------+----------+---------+--------+------+
+| 2002 | Giovanni | Rome    |    200 | 1003 |
+| 2003 | Liu      | SanJose |    200 | 1002 |
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2006 | Clemens  | London  |    100 | 1001 |
+| 2007 | Pereira  | Rome    |    100 | 1004 |
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+19) Give the salespeople’s commissions as percentages instead of decimal numbers. 
+
+mysql> SELECT SNUM,SNAME,CONCAT(COMM,'%') AS 'COMMISSION PERCENT' FROM SALESPEOPLE;
++------+---------+--------------------+
+| SNUM | SNAME   | COMMISSION PERCENT |
++------+---------+--------------------+
+| 1001 | Peel    | 12%                |
+| 1002 | Serres  | 13%                |
+| 1003 | AxelRod | 10%                |
+| 1004 | Motika  | 11%                |
+| 1007 | Rofkin  | 15%                |
+| 1008 | Fran    | 25%                |
++------+---------+--------------------+
+
+20) Find the largest order taken by each salesperson on each date,
+eliminating those Maximum orders, which are less than 3000. 
+
+mysql> SELECT SNUM,MAX(AMT) AS 'LARGEST ORDER',ODATE FROM ORDERS GROUP BY SNUM,ODATE HAVING MAX(AMT)>=3000;
++------+---------------+------------+
+| SNUM | LARGEST ORDER | ODATE      |
++------+---------------+------------+
+| 1002 |       5160.45 | 1990-10-03 |
+| 1001 |          4723 | 1990-10-05 |
+| 1001 |       9891.88 | 1990-10-06 |
++------+---------------+------------+
+
+21) List all the largest orders for October 3, for each salesperson
```

Comment:
what about the salespeople who haven't ordered on oct 3, their largest order should be 0.